### PR TITLE
Fix #486: Do not expose service details when REST client fails

### LINKED ISSUE #486: Do not expose service details when REST client fails
In case HTTP connection to Data Adapter fails during login step, an error message is displayed with  details about Data Adapter service (IP address, port, etc.). Such errors should be handled and a user friendly error message should be displayed. 

Similar issue is most likely present in the Next Step client, too.

### PR DESCRIPTION
**Pull request by romanstrobl**
This pull request contains following improvements related to error handling: 
- Fix #486: Do not expose service details when REST client fails
- Separated HTTP errors from JSON parsing errors
- Use user-friendly error messages instead of messages from Exceptions
- Do not throw Next Step exceptions in Web Flow
- Improved error handling
- Improved error logging

The new error screens are:
1. In case of communication errors with Data Adapter (service is down, a HTTP error occurs, unexpected non-JSON response with HTTP 200 status)

User authentication:
screenshot 2019-03-05 at 15 00 43

SMS authorization:
screenshot 2019-03-05 at 15 03 56

2. In case of communication errors with Next Step server (service is down, a HTTP error occurs, unexpected non-JSON response with HTTP 200 status)

screenshot 2019-03-05 at 15 00 03

Web Flow logs still contain all error details including HTTP status codes or parsing errors.

### CODE REVIEW

File: powerauth-webflow-authentication-mtoken/src/main/java/io/getlime/security/powerauth/lib/webflow/authentication/mtoken/controller/MobileAppApiController.java [hunk 1]
```diff
@@ -142,12 +142,16 @@ protected AuthMethod getAuthMethodName() {
...
-            final List<GetOperationDetailResponse> operationList = getOperationListForUser(userId);
-            if (operationList == null) {
-                // Next step operation failed, return empty operation list
+            List<GetOperationDetailResponse> operationList;
+            Map<String, GetOperationConfigResponse> operationConfigs;
+            try {
+                operationList = getOperationListForUser(userId);
+                operationConfigs = getOperationConfigs(operationList);
+            } catch (AuthStepException e) {
+                // Next step operation list failed, return empty operation list
```
Comment (petrdvorak) [inline powerauth-webflow-authentication-mtoken/src/main/java/io/getlime/security/powerauth/lib/webflow/authentication/mtoken/controller/MobileAppApiController.java]: Should we add logging here?

File: powerauth-webflow-authentication-mtoken/src/main/java/io/getlime/security/powerauth/lib/webflow/authentication/mtoken/controller/MobileAppApiController.java [hunk 2]
```diff
@@ -313,9 +318,10 @@ private boolean verifyActivationId(String activationId, String userId) {
     private boolean isSignatureTypeAllowedForOperation(String operationName, PowerAuthSignatureTypes signatureTypes)  {
 
         // Get configuration for operation with given name
-        final GetOperationConfigResponse operationConfig = getOperationConfig(operationName);
-
-        if (operationConfig == null) {
+        GetOperationConfigResponse operationConfig;
+        try {
+            operationConfig = getOperationConfig(operationName);
+        } catch (AuthStepException e) {
```
Comment (petrdvorak) [inline powerauth-webflow-authentication-mtoken/src/main/java/io/getlime/security/powerauth/lib/webflow/authentication/mtoken/controller/MobileAppApiController.java]: Should we add logging here?

File: powerauth-webflow-authentication-mtoken/src/main/java/io/getlime/security/powerauth/lib/webflow/authentication/mtoken/controller/MobileAppApiController.java (hunk 1 above)
Comment (romanstrobl) [reply powerauth-webflow-authentication-mtoken/src/main/java/io/getlime/security/powerauth/lib/webflow/authentication/mtoken/controller/MobileAppApiController.java]: This is intentional, the Exception would be logged twice.

File: powerauth-webflow-authentication-mtoken/src/main/java/io/getlime/security/powerauth/lib/webflow/authentication/mtoken/controller/MobileAppApiController.java (hunk 2 above)
Comment (romanstrobl) [reply powerauth-webflow-authentication-mtoken/src/main/java/io/getlime/security/powerauth/lib/webflow/authentication/mtoken/controller/MobileAppApiController.java]: This is intentional, the Exception would be logged twice.

Comment (petrdvorak) [reply powerauth-webflow-authentication-mtoken/src/main/java/io/getlime/security/powerauth/lib/webflow/authentication/mtoken/controller/MobileAppApiController.java]: ok

File: powerauth-webflow-authentication-mtoken/src/main/java/io/getlime/security/powerauth/lib/webflow/authentication/mtoken/controller/MobileAppApiController.java (hunk 1 above)
Comment (petrdvorak) [reply powerauth-webflow-authentication-mtoken/src/main/java/io/getlime/security/powerauth/lib/webflow/authentication/mtoken/controller/MobileAppApiController.java]: ok

**Pull request merged**